src/scripts/cronjobs/emailBreachAlerts/subscriptionHandler.spec.ts

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { Logger } from "winston";
+import { MockProcess } from "../../../test/helpers/mockProcess";
+import { mockMessage, MockSubscription } from "../../../test/helpers/pubsub";
+import { mockLogger } from "../../../test/helpers/mockLogger";
+import { MessageSummary, SubscriptionHandler } from "./subscriptionHandler"; // <-- adjust the path
+import { setTimeout } from "timers/promises";
+import { Message, Subscription } from "@google-cloud/pubsub";
+
+describe("SubscriptionHandler", () => {
+  let logger: Logger;
+  let sub: MockSubscription;
+  let proc: MockProcess;
+  let subCloseSpy: jest.SpyInstance<void, []>;
+
+  beforeEach(() => {
+    logger = mockLogger();
+    subCloseSpy = jest.spyOn(MockSubscription.prototype, "close");
+    sub = new MockSubscription();
+    proc = new MockProcess();
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => jest.restoreAllMocks());
+
+  function startHandler(messageFn: (m: Message) => Promise<MessageSummary>) {
+    return new SubscriptionHandler({
+      subscription: sub as unknown as Subscription,
+      messageFn,
+      logger,
+      process: proc as unknown as NodeJS.Process,
+    });
+  }
+
+  it("acks on success=true", async () => {
+    startHandler(async () => ({
+      success: true,
+      errors: 0,
+      notified: 1,
+      skipped: 0,
+    }));
+    const msg = mockMessage({
+      breachName: "Big Breach",
+      hashPrefix: "AA",
+      hashSuffixes: ["11"],
+    });
+
+    sub.emitMessage(msg);
+
+    // allow event loop to flush
+    // no setImmediate in jsdom, so just wait
+    // a short while instead
+    await setTimeout(50);
```

Comment:
Question: Would this also work? I imagine these might start to add up, and timeouts also reek of flakiness.


```suggestion
    await Promise.resolve();
```